.github/workflows/test-apps.yml

```diff
@@ -0,0 +1,42 @@
+name: Test Apps
+
+on:
+  workflow_dispatch: {} # manually
+  schedule:
+    - cron: "0 0 * * *" # nightly
+  pull_request: {} # on pull request
+
+env:
+  DD_ENV: github
+  DD_TAGS: "github_run_id:${{ github.run_id }} github_run_number:${{ github.run_number }}"
+
+jobs:
+  unit-of-work:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          repository: 'DataDog/dd-trace-go'
+
+      - uses: datadog/agent-github-action@v1.3
+        with:
+            api_key: ${{ secrets.DD_TEST_AND_DEMO_API_KEY }}
```

Comment:
Note: This is currently a staging key, but after discussions with Paul, I'd probably like to get a prod key for this instead. Testing in prod is good 😎.

Reply:
We'll do prod + staging. Will close this when done.